### ISSUE TITLE
Powerbi setup failing, and no data sets appear.

### ISSUE THREAD

**Issue by sthoward0914** [ctesta-oneillmsft/asa-vtd#13]
Hello folks.

I am not able to have a successful run on the setup. In the setup, where we have the instructions .\01-environment-setup.ps1, the Powerbi setup seems to fail. as a result, when I get to the part of looking for the datasets in Powerbi, they're not there.

So far as I can tell, everything is okay until that point. this is the steps that are causing the error, and the errors they are causing.

Uploading PowerBI Report 1-CDP Vision Demo
Invoke-RestMethod : The remote server returned an error: (404) Not Found.
At C:\labfiles\synapse-in-a-day-deployment\artifacts\environment-setup\solliance-synapse-automation\solliance-synapse-automation.p
sm1:1664 char:15
+ ...   $result = Invoke-RestMethod  -Uri $url -Method POST -Body $bodyLine ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (System.Net.HttpWebRequest:HttpWebRequest) [Invoke-RestMethod], WebException
    + FullyQualifiedErrorId : WebCmdletWebResponseException,Microsoft.PowerShell.Commands.InvokeRestMethodCommand
 
Invoke-RestMethod : The remote server returned an error: (404) Not Found.
At C:\labfiles\synapse-in-a-day-deployment\artifacts\environment-setup\solliance-synapse-automation\solliance-synapse-automation.p
sm1:1687 char:15
+ ...   $result = Invoke-RestMethod  -Uri $url -Method GET -Headers @{ Auth ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (System.Net.HttpWebRequest:HttpWebRequest) [Invoke-RestMethod], WebException
    + FullyQualifiedErrorId : WebCmdletWebResponseException,Microsoft.PowerShell.Commands.InvokeRestMethodCommand
 
Uploading 2-Billion Rows Demo Report
Uploading PowerBI Report 2-Billion Rows Demo
Invoke-RestMethod : The remote server returned an error: (404) Not Found.
At C:\labfiles\synapse-in-a-day-deployment\artifacts\environment-setup\solliance-synapse-automation\solliance-synapse-automation.p
sm1:1664 char:15
+ ...   $result = Invoke-RestMethod  -Uri $url -Method POST -Body $bodyLine ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (System.Net.HttpWebRequest:HttpWebRequest) [Invoke-RestMethod], WebException
    + FullyQualifiedErrorId : WebCmdletWebResponseException,Microsoft.PowerShell.Commands.InvokeRestMethodCommand
 
Invoke-RestMethod : The remote server returned an error: (404) Not Found.
At C:\labfiles\synapse-in-a-day-deployment\artifacts\environment-setup\solliance-synapse-automation\solliance-synapse-automation.p
sm1:1687 char:15
+ ...   $result = Invoke-RestMethod  -Uri $url -Method GET -Headers @{ Auth ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (System.Net.HttpWebRequest:HttpWebRequest) [Invoke-RestMethod], WebException
    + FullyQualifiedErrorId : WebCmdletWebResponseException,Microsoft.PowerShell.Commands.InvokeRestMethodCommand
 
Uploading 1-Phase2 CDP Vision Demo Report
Uploading PowerBI Report 1-Phase2 CDP Vision Demo
Invoke-RestMethod : The remote server returned an error: (404) Not Found.
At C:\labfiles\synapse-in-a-day-deployment\artifacts\environment-setup\solliance-synapse-automation\solliance-synapse-automation.p
sm1:1664 char:15
+ ...   $result = Invoke-RestMethod  -Uri $url -Method POST -Body $bodyLine ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (System.Net.HttpWebRequest:HttpWebRequest) [Invoke-RestMethod], WebException
    + FullyQualifiedErrorId : WebCmdletWebResponseException,Microsoft.PowerShell.Commands.InvokeRestMethodCommand
 
Invoke-RestMethod : The remote server returned an error: (404) Not Found.
At C:\labfiles\synapse-in-a-day-deployment\artifacts\environment-setup\solliance-synapse-automation\solliance-synapse-automation.p
sm1:1687 char:15
+ ...   $result = Invoke-RestMethod  -Uri $url -Method GET -Headers @{ Auth ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (System.Net.HttpWebRequest:HttpWebRequest) [Invoke-RestMethod], WebException
    + FullyQualifiedErrorId : WebCmdletWebResponseException,Microsoft.PowerShell.Commands.InvokeRestMethodCommand
 
Uploading Dashboard-Images Report
Uploading PowerBI Report Dashboard-Images
Invoke-RestMethod : The remote server returned an error: (404) Not Found.
At C:\labfiles\synapse-in-a-day-deployment\artifacts\environment-setup\solliance-synapse-automation\solliance-synapse-automation.p
sm1:1664 char:15
+ ...   $result = Invoke-RestMethod  -Uri $url -Method POST -Body $bodyLine ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (System.Net.HttpWebRequest:HttpWebRequest) [Invoke-RestMethod], WebException
    + FullyQualifiedErrorId : WebCmdletWebResponseException,Microsoft.PowerShell.Commands.InvokeRestMethodCommand
 
Invoke-RestMethod : The remote server returned an error: (404) Not Found.
At C:\labfiles\synapse-in-a-day-deployment\artifacts\environment-setup\solliance-synapse-automation\solliance-synapse-automation.p
sm1:1687 char:15
+ ...   $result = Invoke-RestMethod  -Uri $url -Method GET -Headers @{ Auth ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (System.Net.HttpWebRequest:HttpWebRequest) [Invoke-RestMethod], WebException
    + FullyQualifiedErrorId : WebCmdletWebResponseException,Microsoft.PowerShell.Commands.InvokeRestMethodCommand
 
Create PowerBI linked service asapowerbisthoward
Create-PowerBILinkedService : Cannot bind argument to parameter 'WorkspaceId' because it is an empty string.
At C:\labfiles\synapse-in-a-day-deployment\artifacts\environment-setup\automation\01-environment-setup.ps1:1158 char:131
+ ... h -WorkspaceName $workspaceName -Name $powerBIName -WorkspaceId $wsid
+                                                                     ~~~~~
    + CategoryInfo          : InvalidData: (:) [Create-PowerBILinkedService], ParameterBindingValidationException
    + FullyQualifiedErrorId : ParameterArgumentValidationErrorEmptyStringNotAllowed,Create-PowerBILinkedService
 
Setting PowerBI Report Data Connections
Setting database connection for 1-CDP Vision Demo
No report found called 
Setting database connection for 2-Billion Rows Demo
No report found called 
Setting database connection for 1-Phase2 CDP Vision Demo
No report found called

**Comment by givenscj** [comment]
HI there, what is the name of the resource group you are using to support your setup?

**Comment by sthoward0914** [comment]
The name of the resource group is synapse-in-a-day-demos (Copy/pasted so I wouldn't make a typo).

I was going to try to get into it tomorrow to try to fix it, but if you have something simple, I'd love to know it was my mistake and I can continue. :)

**Comment by joelhulen** [comment]
I have merged Chris's changes. Can you please update your local copy of the repo and try again?

**Comment by sthoward0914** [comment]
I get a little different error this time. I'm actually a bit surprised that anything still has any dependencies on Internet Explorer. 

I have a Dr. appointment this morning, then I'll try to make this work.

Error message:

Starting PowerBI Artifact Provisioning
Invoke-WebRequest : The response content cannot be parsed because the Internet Explorer engine is not available, or Internet 
Explorer's first-launch configuration is not complete. Specify the UseBasicParsing parameter and try again. 
At C:\labfiles\synapse-in-a-day-deployment\artifacts\environment-setup\solliance-synapse-automation\solliance-synapse-automation.p
sm1:1621 char:15
+ ...   $result = Invoke-WebRequest -Uri $url -Method GET -ContentType "app ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotImplemented: (:) [Invoke-WebRequest], NotSupportedException
    + FullyQualifiedErrorId : WebCmdletIEDomNotSupportedException,Microsoft.PowerShell.Commands.InvokeWebRequestCommand
 
Cannot index into a null array.
At C:\labfiles\synapse-in-a-day-deployment\artifacts\environment-setup\solliance-synapse-automation\solliance-synapse-automation.p
sm1:1622 char:5
+     $homeCluster = $result.Headers["home-cluster-uri"]
+     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [], RuntimeException
    + FullyQualifiedErrorId : NullArray
 
Invoke-RestMethod : The remote server returned an error: (400) Bad Request.
At C:\labfiles\synapse-in-a-day-deployment\artifacts\environment-setup\solliance-synapse-automation\solliance-synapse-automation.p
sm1:1614 char:15
+ ...   $result = Invoke-RestMethod -Uri $url -Method POST -Body $body -Con ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (System.Net.HttpWebRequest:HttpWebRequest) [Invoke-RestMethod], WebException
    + FullyQualifiedErrorId : WebCmdletWebResponseException,Microsoft.PowerShell.Commands.InvokeRestMethodCommand
 
Invoke-RestMethod : The remote server returned an error: (404) Not Found.
At C:\labfiles\synapse-in-a-day-deployment\artifacts\environment-setup\solliance-synapse-automation\solliance-synapse-automation.p
sm1:1725 char:15
+ ...   $result = Invoke-RestMethod -Uri $url -Method POST -body $post -Con ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (System.Net.HttpWebRequest:HttpWebRequest) [Invoke-RestMethod], WebException
    + FullyQualifiedErrorId : WebCmdletWebResponseException,Microsoft.PowerShell.Commands.InvokeRestMethodCommand
 
Uploading PowerBI Reports
0
1
2
3
Uploading 1-CDP Vision Demo Report
Uploading PowerBI Report 1-CDP Vision Demo
Invoke-RestMethod : The remote server returned an error: (404) Not Found.
At C:\labfiles\synapse-in-a-day-deployment\artifacts\environment-setup\solliance-synapse-automation\solliance-synapse-automation.p
sm1:1664 char:15
+ ...   $result = Invoke-RestMethod  -Uri $url -Method POST -Body $bodyLine ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (System.Net.HttpWebRequest:HttpWebRequest) [Invoke-RestMethod], WebException
    + FullyQualifiedErrorId : WebCmdletWebResponseException,Microsoft.PowerShell.Commands.InvokeRestMethodCommand
 
Invoke-RestMethod : The remote server returned an error: (404) Not Found.
At C:\labfiles\synapse-in-a-day-deployment\artifacts\environment-setup\solliance-synapse-automation\solliance-synapse-automation.p
sm1:1687 char:15
+ ...   $result = Invoke-RestMethod  -Uri $url -Method GET -Headers @{ Auth ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (System.Net.HttpWebRequest:HttpWebRequest) [Invoke-RestMethod], WebException
    + FullyQualifiedErrorId : WebCmdletWebResponseException,Microsoft.PowerShell.Commands.InvokeRestMethodCommand
 
Uploading 2-Billion Rows Demo Report
Uploading PowerBI Report 2-Billion Rows Demo
Invoke-RestMethod : The remote server returned an error: (404) Not Found.
At C:\labfiles\synapse-in-a-day-deployment\artifacts\environment-setup\solliance-synapse-automation\solliance-synapse-automation.p
sm1:1664 char:15
+ ...   $result = Invoke-RestMethod  -Uri $url -Method POST -Body $bodyLine ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (System.Net.HttpWebRequest:HttpWebRequest) [Invoke-RestMethod], WebException
    + FullyQualifiedErrorId : WebCmdletWebResponseException,Microsoft.PowerShell.Commands.InvokeRestMethodCommand
 
Invoke-RestMethod : The remote server returned an error: (404) Not Found.
At C:\labfiles\synapse-in-a-day-deployment\artifacts\environment-setup\solliance-synapse-automation\solliance-synapse-automation.p
sm1:1687 char:15
+ ...   $result = Invoke-RestMethod  -Uri $url -Method GET -Headers @{ Auth ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (System.Net.HttpWebRequest:HttpWebRequest) [Invoke-RestMethod], WebException
    + FullyQualifiedErrorId : WebCmdletWebResponseException,Microsoft.PowerShell.Commands.InvokeRestMethodCommand
 
Uploading 1-Phase2 CDP Vision Demo Report
Uploading PowerBI Report 1-Phase2 CDP Vision Demo
Invoke-RestMethod : The remote server returned an error: (404) Not Found.
At C:\labfiles\synapse-in-a-day-deployment\artifacts\environment-setup\solliance-synapse-automation\solliance-synapse-automation.p
sm1:1664 char:15
+ ...   $result = Invoke-RestMethod  -Uri $url -Method POST -Body $bodyLine ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (System.Net.HttpWebRequest:HttpWebRequest) [Invoke-RestMethod], WebException
    + FullyQualifiedErrorId : WebCmdletWebResponseException,Microsoft.PowerShell.Commands.InvokeRestMethodCommand
 
Invoke-RestMethod : The remote server returned an error: (404) Not Found.
At C:\labfiles\synapse-in-a-day-deployment\artifacts\environment-setup\solliance-synapse-automation\solliance-synapse-automation.p
sm1:1687 char:15
+ ...   $result = Invoke-RestMethod  -Uri $url -Method GET -Headers @{ Auth ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (System.Net.HttpWebRequest:HttpWebRequest) [Invoke-RestMethod], WebException
    + FullyQualifiedErrorId : WebCmdletWebResponseException,Microsoft.PowerShell.Commands.InvokeRestMethodCommand
 
Uploading Dashboard-Images Report
Uploading PowerBI Report Dashboard-Images
Invoke-RestMethod : The remote server returned an error: (404) Not Found.
At C:\labfiles\synapse-in-a-day-deployment\artifacts\environment-setup\solliance-synapse-automation\solliance-synapse-automation.p
sm1:1664 char:15
+ ...   $result = Invoke-RestMethod  -Uri $url -Method POST -Body $bodyLine ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (System.Net.HttpWebRequest:HttpWebRequest) [Invoke-RestMethod], WebException
    + FullyQualifiedErrorId : WebCmdletWebResponseException,Microsoft.PowerShell.Commands.InvokeRestMethodCommand
 
Invoke-RestMethod : The remote server returned an error: (404) Not Found.
At C:\labfiles\synapse-in-a-day-deployment\artifacts\environment-setup\solliance-synapse-automation\solliance-synapse-automation.p
sm1:1687 char:15
+ ...   $result = Invoke-RestMethod  -Uri $url -Method GET -Headers @{ Auth ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (System.Net.HttpWebRequest:HttpWebRequest) [Invoke-RestMethod], WebException
    + FullyQualifiedErrorId : WebCmdletWebResponseException,Microsoft.PowerShell.Commands.InvokeRestMethodCommand
 
Create PowerBI linked service asapowerbisthoward
Create-PowerBILinkedService : Cannot bind argument to parameter 'WorkspaceId' because it is an empty string.
At C:\labfiles\synapse-in-a-day-deployment\artifacts\environment-setup\automation\01-environment-setup.ps1:1167 char:131
+ ... h -WorkspaceName $workspaceName -Name $powerBIName -WorkspaceId $wsid
+                                                                     ~~~~~
    + CategoryInfo          : InvalidData: (:) [Create-PowerBILinkedService], ParameterBindingValidationException
    + FullyQualifiedErrorId : ParameterArgumentValidationErrorEmptyStringNotAllowed,Create-PowerBILinkedService
 
Setting PowerBI Report Data Connections
Setting database connection for 1-CDP Vision Demo
No report found called 
Setting database connection for 2-Billion Rows Demo
No report found called 
Setting database connection for 1-Phase2 CDP Vision Demo
No report found called

**Comment by sthoward0914** [comment]
In order to get anything to work, I had to create the powerbi workspace manually because that command to create it continued to return an error from invoke-restrequest. Since the setup script takes to long to run, I extracted out the powerbi setup portion, just defined variables needed, and tried to run that. It worked except for missing credentials in datasets (which I think also returned an error from invoke-restrequest). Once I resolved those, they worked.
